### PR TITLE
Login as claim

### PR DESCRIPTION
**Pull request by benrudolph**
@proteusvacuum @wpride @ctsims i checked on the server when trying to claim the case in https://manage.dimagi.com/default.asp?253902 that the session said i had already claimed it. i realized that we were storing the session id of the web user who claimed the case, not the restore user. so if a web user was logged in as a different user and claimed a case, and then logged in as a different user and claimed that same case, it would fail. see test case for example

### CODE REVIEW

**Review by snopoke — APPROVED** [review]

**Comment by wpride** [comment]
@benrudolph 💵 thanks man

File: corehq/apps/ota/tests/test_search_claim_endpoints.py [hunk 1]
```diff
@@ -141,6 +144,43 @@ def test_claim_restore_as(self):
         claim_case = CaseAccessors(DOMAIN).get_case(claim_ids[0])
         self.assertEqual(claim_case.owner_id, other_user._id)
 
+    def test_claim_restore_as_proper_cache(self):
```
Comment (proteusvacuum) [inline corehq/apps/ota/tests/test_search_claim_endpoints.py]: probably not a huge issue, but the other tests here `@run_with_all_backends`

**Comment by proteusvacuum** [comment]
Thanks @benrudolph

File: corehq/apps/ota/tests/test_search_claim_endpoints.py (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/ota/tests/test_search_claim_endpoints.py]: yeah i thought about it, but decided i wasn't testing that functionality with this test and didn't want to add the overhead

**Pull request merged**